base/intfuncs.jl

```diff
@@ -605,8 +605,8 @@ function bin(x::Unsigned, pad::Integer, neg::Bool)
     i = neg + max(pad,sizeof(x)<<3-leading_zeros(x))
     a = StringVector(i)
     while i > neg
-        @inbounds a[i] = 48+(x&0x1)
-        x >>= 1
+        @inbounds a[i] = (0x30+(x&0x1)::Unsigned)::Unsigned
+        x = (x >> 1)::Unsigned
```

Comment:
```suggestion
        @inbounds a[i] = 0x30 + ((x % UInt8)::UInt8 & 0x1)
        x >>= 0x1
```
I think this may clarify the intent. As mentioned above, `(pad % Int)::Int` may be safer.